File: src/exactlyelectrical.py
a, b = [int(d) for d in input().split()]
c, d = [int(d) for d in input().split()]
t = int(input())
m = abs(a - c) + abs(b - d)
print("Y" if m <= t and (m - t) % 2 == 0 else "N")

File: src/exactlyelectrical.lua
a, b, c, d, t = io.read("*n", "*n", "*n", "*n", "*n")
m = math.abs(a - c) + math.abs(b - d)
if m <= t and (m - t) % 2 == 0 then
    print("Y")
else
    print("N")
end
